simulation.cpp

```diff
@@ -244,7 +245,13 @@ void assign_inputs(simulation &s)
 
 void assign_new_inputs(simulation &s)
 {
-    s.update_inputs(create_inputs(s));
+    std::vector<double> new_inputs = create_inputs(s);
+
+    if(s.get_input().size() > 1){
```

Comment:
Nice check, but this `if` statement should go inside the body of `create_inputs`. It is really the job of `create_inputs` to provide simulation with the correct inputs. `assign_new_inputs()` should not care, should just assign.